test/routing/ArrivalsRouterSpec.scala

```diff
@@ -48,75 +50,84 @@ class ArrivalsRouterSpec extends AnyFreeSpec with Matchers with OptionValues wit
     new FakeV2ArrivalsController()
   )
 
-  "when creating a Arrival Notification" - {
-    "with accept header set to application/vnd.hmrc.2.0+json (version two)" - {
-
-      val arrivalsHeaders = FakeHeaders(
-        Seq(HeaderNames.ACCEPT -> "application/vnd.hmrc.2.0+json", HeaderNames.CONTENT_TYPE -> "application/xml")
-      )
-
-      "must route to the v2 controller and return Accepted when successful" in {
-
-        val request =
-          FakeRequest(method = "POST", uri = routes.ArrivalsRouter.createArrivalNotification().url, body = <test></test>, headers = arrivalsHeaders)
-        val result = call(sut.createArrivalNotification(), request)
-
-        status(result) mustBe ACCEPTED
-        contentAsJson(result) mustBe Json.obj("version" -> 2) // ensure we get the unique value to verify we called the fake action
-      }
-
+  val id = Gen.long
+    .map {
+      l: Long =>
+        f"${BigInt(abs(l))}%016x"
     }
-
-    Seq(None, Some("application/vnd.hmrc.1.0+json"), Some("text/html"), Some("application/vnd.hmrc.1.0+xml"), Some("text/javascript")).foreach {
-      acceptHeaderValue =>
-        val acceptHeader = acceptHeaderValue
-          .map(
-            header => Seq(HeaderNames.ACCEPT -> header)
-          )
-          .getOrElse(Seq.empty)
-        val arrivalHeaders = FakeHeaders(acceptHeader ++ Seq(HeaderNames.CONTENT_TYPE -> "application/xml"))
-        val withString = acceptHeaderValue
-          .getOrElse("nothing")
-        s"with accept header set to $withString" - {
-
-          "must route to the v1 controller and return Accepted when successful" in {
-
-            val request =
-              FakeRequest(method = "POST", uri = routes.ArrivalsRouter.createArrivalNotification().url, body = <test></test>, headers = arrivalHeaders)
-            val result = call(sut.createArrivalNotification(), request)
-
-            status(result) mustBe ACCEPTED
-            contentAsJson(result) mustBe Json.obj("version" -> 1) // ensure we get the unique value to verify we called the fake action
-          }
+    .sample
+    .get
+
+  for (
+    callAndSutAndStatus <- Seq(
+      (routes.ArrivalsRouter.createArrivalNotification(), sut.createArrivalNotification(), ACCEPTED),
+      (routes.ArrivalsRouter.getArrival(id), sut.getArrival(id), OK),
+      (routes.ArrivalsRouter.getArrivalsForEori(), sut.getArrivalsForEori(), OK)
+    )
```

Comment:
I'm torn about this.

It's a clever way to reduce code (and I am indeed the sort of person who likes this sort of thing!), but it's not obvious what it's doing and why. Someone looking to add another endpoint that does version checks like this would probably get confused and/or write their own test that does exactly this because they don't understand it -- especially as it's outside of any test description block.

I think, for readability's sake, do something like this:

```scala
"routing to version 2 with a version 2 accept header" - {
    // second parameter is call by name so it only evaluates when we want it to
    def executeTest(callValue: (type), sutValue: => Future[Result], expectedStatus: Int) = {
        // test
    }

   "when creating an arrival notification" in executeTest(...)
   "when getting an arrival" in executeTest(...)
   "when getting arrivals for a given enrolment EORI" in executeTest(...)
}
```

Reply:
That's a good point Dan, thanks for the suggestion